### DISCUSSION TITLE
How does it sync the filetree

### DISCUSSION THREAD

**Discussion by sarmong** [nvim-tree/nvim-tree.lua#1656] [General]
I remember adding `reload_on_bufenter` option sometime ago. Now that I updated, I noticed that nvim tree updates the tree when you create a new file even without entering a buffer and with this option off.
Is this some new feature I missed?

Hence the question - maybe `reload_on_bufenter` is now redundant option?

**Comment by alex-courtis** [comment]
> I remember adding `reload_on_bufenter` option sometime ago. Now that I updated, I noticed that nvim tree updates the tree when you create a new file even without entering a buffer and with this option off. Is this some new feature I missed?
> 
> Hence the question - maybe `reload_on_bufenter` is now redundant option?

nvim-tree uses file system watchers to update parts of the tree on changes. This was added several months ago and was made the default behaviour shortly after.

`reload_on_bufenter` is longer needed and will be silently removed soon: #1651

**Reply by sarmong** [reply]
That is a great feature!
